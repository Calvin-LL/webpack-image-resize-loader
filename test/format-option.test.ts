import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe('"format" option', () => {
  it("should work with jpeg", async () => {
    const compiler = getCompiler({
      size: {
        width: 10,
      },
      format: "jpeg",
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it("should work with png", async () => {
    const compiler = getCompiler({
      size: {
        width: 10,
      },
      format: "png",
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it("should work with webp", async () => {
    const compiler = getCompiler({
      size: {
        width: 10,
      },
      format: "webp",
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it("should work with tiff", async () => {
    const compiler = getCompiler({
      size: {
        width: 10,
      },
      format: "tiff",
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });
});
