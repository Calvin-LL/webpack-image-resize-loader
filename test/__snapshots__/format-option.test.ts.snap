// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"format" option should work with jpeg: result 1`] = `"e9cac5dc1b3f4b537ff58229490ade0c.jpg"`;

exports[`"format" option should work with png: result 1`] = `"6199cbefbf619abc909266f78352da1d.jpg"`;

exports[`"format" option should work with tiff: result 1`] = `"fd1e1edb6b7d9cb140d2b521aa60035f.jpg"`;

exports[`"format" option should work with webp: result 1`] = `"1fd44d17eabb4e5e4cb666a23c08fe4f.jpg"`;
